thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashParams.js

```diff
@@ -81,9 +91,38 @@ HashParams.prototype = {
       this.controllerNameToParamNamesMap[this.ANOMALIES_CONTROLLER] = paramNamesToDefaultValuesMap;
 
       // analysis
-
+      paramNamesToDefaultValuesMap = {};
+      paramNamesToDefaultValuesMap[this.TAB] = constants.TAB_ANALYSIS;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_CURRENT_START] = null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_CURRENT_END]= null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_METRIC_ID]= null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_BASELINE_START] = null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_BASELINE_END] = null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_FILTERS] = null;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_GRANULARITY] = constants.DEFAULT_ANALYSIS_GRANULARITY;
+      paramNamesToDefaultValuesMap[this.ANALYSIS_DIMENSION] = constants.DEFAULT_ANALYSIS_DIMENSION;
+      this.controllerNameToParamNamesMap[this.ANALYSIS_CONTROLLER] = paramNamesToDefaultValuesMap;
 
       console.log('hash Params init');
       console.log(this.controllerNameToParamNamesMap);
-    }
-}
+    },
+
+    isSame(key, currentValue, newValue) {
+      switch(key){
+        case this.ANALYSIS_CURRENT_START:
+        case this.ANALYSIS_CURRENT_END :
+        case this.ANALYSIS_BASELINE_START :
+        case this.ANALYSIS_BASELINE_END :
+        case this.ANOMALIES_START_DATE :
+        case this.ANOMALIES_END_DATE:
+          return currentValue.isSame(newValue, 'day');
+          break;
+        case this.ANALYSIS_FILTERS:
+          return JSON.stringify(currentValue) == newValue;
```

Comment:
Should this be an identity `===` comparison? This is preferred as a default except you're intentionally seeking the implicit type conversion with `==`, in that case it is better to be explicit with a type conversion operation e.g. using `String()` or concatenating with an empty string literal `'' + ...`